### ISSUE TITLE
Anyone used it with material ui?

### ISSUE THREAD

**Issue by thesabbir** [reapp/reapp#88]
Can I use http://material-ui.com with this without getting much trouble?

**Comment by mbrookes** [comment]
You can, but given that a large part of what reapp offers (outside the build system) is reapp-ui, you'll be missing out. :) Better if you were able to apply material styling to reapp components (as a new theme for example).

**Comment by mbrookes** [comment]
Incidentally, I've been working on a project with Material-UI and Meteor, and whilst I'm loving Meteor, Material-UI really made me appreciate just how strong reapp-ui is. The API for Material-UI components is all over the place - inconsistent, and brittle. And good luck trying to animate with even basic transitions, never mind the trying to reproduce the touch friendly animations that reapp provides.

Honestly, the experience has given me a whole new appreciation for reapp.

**Comment by timuric** [comment]
@thesabbir Yep latest version works just fine. The setup is very simple, I have it running by just adding these two lines to the app.js:

`let injectTapEventPlugin = require('react-tap-event-plugin');`

`injectTapEventPlugin();`

**Comment by timuric** [comment]
@thesabbir One think to keep in mind when using material-ui, is that Reapp uses `display:flex` globally, which brakes some of the layout of some material-ui components. But luckily you can pass custom style props to MUI components to fix it.

**Comment by mbrookes** [comment]
@timuric - which bits of reapp do you use with material & why? Views? Might have to rethink how much of the wheel I'm having to reinvent using Material-UI by itself...

**Comment by timuric** [comment]
@mbrookes Yep, from Reapp I mostly use views for transitions, and sometimes List component. Material-ui has quite rich selection of elements, and it is handy for both desktop and mobile apps. The tradeoff is of course the overall feel of the application, if you build something exclusively for iOS then Reapp components fit better. At the moment I am experimenting with the two at the same time, however in the future I would probably would use them separately or try to align them into one package :)

**Comment by mbrookes** [comment]
@timuric I've gone a slightly different route: I started with Reapp / Flux / PouchDB on the client synced with CouchDB on the server with Rails for the admin / web view, but the Reapp bug that breaks things on orientation change (#24) was a show-stopper, and with Reapp development on hold, couldn't risk that it might never get fixed. 

So I switched to Material-UI on the client, and Meteor (GroundDB / Mondo) for client-server and build system.  The biggest challenge has been with nested views (and remembering view state) and transition animations (which is just CSS transitions like TouchstoneJS uses - no realtime interactivity). 

Even getting the CSS for a fixed header / footer with scrollable content right took a weekend of head-scratching (CSS is not my speciality!). Stupid simple things that take a ton of time away from actually coding the app. It was that which made me appreciate how much Reapp takes care of for you.

If that bug was fixed, I'd be interested to try a Reapp Material-UI mashup. How have you got around it?

**Comment by timuric** [comment]
@mbrookes it works but the biggest problem though is that Reapp has `display: flex` set globally for elements on the page which brakes some material-ui components. One possible workaround is to create a css class like so : 

```ccc
.forceBlock > * {
  display: block;
} 

and apply it to broken MUI components. It's quite ugly, but it is an easy workaround. Another concern is that you would end up having two themes in your project. It is definitely usable but not elegant. In any case I think that it would be a good idea to remove global flexbox from Reapp, to avoid conflicts with external libraries.

**Comment by mbrookes** [comment]
Hmm - I'm using MUI with flex containers, and I can't say I've had any significant problems, but I might just not be using the components you've had problems with. Anything in particular?

IMHO if there are any issues with flex, it should be up to MUI (or other component library) to fix, rather than devs have to avoid flex - it's a (draft) web standard after all, and supported by all modern browsers.

**Comment by timuric** [comment]
@mbrookes For example component `Checkbox` aligns weirdly because it's children use floats  http://d.pr/i/1c95Y 

Flexbox is great but I am afraid there are many UI components on the web that are build with box model in order to support IE9 such as [elemental](http://elemental-ui.com/), [react-boostrap](https://react-bootstrap.github.io/) or other existing components. I guess it is not an easy choice which way to go :)

**Comment by mbrookes** [comment]
Ah, haven't needed that one. Building components that work in IE9 is fine, so long as they also play nice with flex. :wink:
